qiskit/primitives/containers/observables_array.py

```diff
@@ -273,6 +281,73 @@ def coerce(cls, observables: ObservablesArrayLike) -> ObservablesArray:
             return observables
         return cls(observables)
 
+    def equivalent(self, other: ObservablesArray, tol: float = 1e-08):
+        """Return True if arrays are equal"""
+        if self.num_qubits != other.num_qubits or self.shape != other.shape:
+            return False
+
+        arr1 = np.atleast_1d(self._array).flat
+        arr2 = np.atleast_1d(other._array).flat
+
+        for obs1, obs2 in zip(arr1, arr2):
```

Comment:
ravel does what you want here
```suggestion
        for obs1, obs2 in zip(self._array.ravel(), other._array.ravel()):
```